tornado/options.py

```diff
@@ -319,7 +319,17 @@ def parse_config_file(self, path, final=True):
         for name in config:
             normalized = self._normalize_name(name)
             if normalized in self._options:
-                self._options[normalized].set(config[name])
+                option = self._options[normalized]
+                if option.multiple:
+                    if not isinstance(config[name], list):
+                        raise Error("Option %r is required to be a list of %s" %
+                                    (option.name, option.type.__name__))
+
+                if type(config[name]) == str and option.type != str:
```

Comment:
What about lists of strings? I think if the value is a list we should loop over the list and use this logic on each element.

Reply:
Doing this turned out to be tricky, but I added a clause that allows the comma-separated string format to be used in this case.